File: 283. Move Zeroes/main.go
package main

import "fmt"

func main() {
	nums := []int{0, 1, 0, 3, 12}
	moveZeroes(nums)
	fmt.Println(nums)
}

func moveZeroes(nums []int) {

	i, j := 0, 0

	for j < len(nums) {

		if nums[i] == 0 && nums[j] != 0 {
			nums[i], nums[j] = nums[j], nums[i]
		}

		if nums[i] != 0 {
			i++
		}
		j++
	}
}

File: 283. Move Zeroes/main.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        slow = 0
        fast = 0
        while fast < len(nums) and slow < len(nums):
            while fast < len(nums) and nums[fast] == 0:
                fast += 1
            if fast < len(nums):
                nums[slow], nums[fast] = nums[fast], nums[slow]
                slow += 1
            fast += 1
